contracts/minter-suite/Minters/MinterSetPriceV5Merkle.sol

```diff
@@ -0,0 +1,684 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+// Created By: Art Blocks Inc.
+
+import "../../interfaces/0.8.x/IGenArt721CoreContractV3_Base.sol";
+import "../../interfaces/0.8.x/IDelegationRegistry.sol";
+import "../../interfaces/0.8.x/IFilteredMinterV3.sol";
+import "../../interfaces/0.8.x/IMinterFilterV1.sol";
+import "../../interfaces/0.8.x/IFilteredSharedProjectMaxInvocationsLimit.sol";
+import "../../interfaces/0.8.x/IFilteredSharedMerkle.sol";
+
+import "../../libs/0.8.x/MerkleLib.sol";
+import "../../libs/0.8.x/SplitFundsLib.sol";
+
+import "@openzeppelin-4.5/contracts/security/ReentrancyGuard.sol";
+
+pragma solidity 0.8.17;
+
+/**
+ * @title Filtered Minter contract that allows tokens to be minted with ETH.
+ * This is designed to be used with GenArt721CoreContractV3 flagship or
+ * engine contracts.
+ * @author Art Blocks Inc.
+ * @notice Privileged Roles and Ownership:
+ * This contract is designed to be managed, with limited powers.
+ * Privileged roles and abilities are controlled by the project's artist, which
+ * can be modified by the core contract's Admin ACL contract. Both of these
+ * roles hold extensive power and can modify minter details.
+ * Care must be taken to ensure that the admin ACL contract and artist
+ * addresses are secure behind a multi-sig or other access control mechanism.
+ * ----------------------------------------------------------------------------
+ * The following functions are restricted to a project's artist:
+ * - updatePricePerTokenInWei
+ * - setProjectMaxInvocations
+ * - manuallyLimitProjectMaxInvocations
+ * ----------------------------------------------------------------------------
+ * Additional admin and artist privileged roles may be described on other
+ * contracts that this minter integrates with.
+ */
+contract MinterSetPriceV5Merkle is
+    ReentrancyGuard,
+    IFilteredMinterV3,
+    IFilteredSharedMerkle,
+    IFilteredSharedProjectMaxInvocationsLimit
+{
+    /// Minter filter address this minter interacts with
+    address public immutable minterFilterAddress;
+
+    /// Minter filter this minter may interact with.
+    IMinterFilterV1 private immutable minterFilter;
+
+    /// minterType for this minter
+    string public constant minterType = "MinterSetPriceV5";
+
+    /// minter version for this minter
+    string public constant minterVersion = "v5.0.0";
+
+    uint256 constant ONE_MILLION = 1_000_000;
+
+    /// contractAddress => projectId => project config
+    mapping(address => mapping(uint256 => MerkleLib.ProjectConfig))
+        public projectConfig;
```

Comment:
as noted above, we should figure out where we want to define projectConfig and update here accordingly